clueless/models.py

```diff
@@ -248,11 +257,36 @@ def startGame(self):
         #TODO: implement this method
         pass
 
+    def isUserInGame(self, user):
+        """
+        Checks whether a user is in the game or not
+        :param user:
+        :return: True if user can join game
+        """
+        try:
+            player = Player.objects.get(currentGame__id=self.id, user__id=user.id)
+        except:
+            return False
+        return True
+
+    def isCharacterInGame(self, user):
+        """
+        Checks whether the character is being used in the game
+        :param user:
+        :return:
+        """
+        #TODO: implement this method
+        return False
+
     def addPlayer(self, player):
         """
         Adds a player to the game
         :param player: Player to be added
         """
+        if self.isUserInGame(player.user):
+            raise RuntimeError("User is already a player in this game")
+        elif self.isCharacterInGame(player.character):
+            raise RuntimeError("Character is already in use")
         player.currentGame = self
```

Comment:
Agreed- plan to implement that.  But if they load the page, then somebody joins, somebody could take that character between when their page loads and when they submit, hence the need for a check